web/src/unread.js

```diff
@@ -274,8 +274,24 @@ class UnreadTopicCounter {
                 continue;
             }
 
-            res.stream_count.set(stream_id, this.get_stream_count(stream_id));
-            res.stream_unread_messages += res.stream_count.get(stream_id).unmuted_count;
+            if (include_per_topic_count) {
+                const topic_unread = new Map();
+                let stream_count = 0;
+                for (const [topic, msgs] of per_stream_bucketer) {
+                    const topic_count = msgs.size;
+                    topic_unread.set(topic, topic_count);
+                    stream_count += topic_count;
+                }
+
+                // TODO: These don't agree with the else clause in how
+                // they handle muted streams/topics, and that's
+                // probably a bug.
+                res.stream_count.set(stream_id, topic_unread);
+                res.stream_unread_messages += stream_count;
```

Comment:
So I think the `stream_count` calculations here may differ between the two conditions here; the else clause here considers muting/unmuting. I think this means the inbox view can differ from the left sidebar in what unread counts it displays; this is probably a bug. But as it's a bug localized to the inbox view, I think we can fix it via a follow-up?

```
    get_stream_count(stream_id) {                                                                       
        const per_stream_bucketer = this.bucketer.get_bucket(stream_id);                                
                                                                                                        
        if (!per_stream_bucketer) {                                                                     
            return 0;                                                                                   
        }                                                                                               
                                                                                                        
        const sub = sub_store.get(stream_id);                                                           
        let unmuted_count = 0;                                                                          
        let muted_count = 0;                                                                            
        for (const [topic, msgs] of per_stream_bucketer) {                                              
            const topic_count = msgs.size;                                                              
                                                                                                        
            if (user_topics.is_topic_unmuted(stream_id, topic)) {                                       
                unmuted_count += topic_count;                                                           
            } else if (user_topics.is_topic_muted(stream_id, topic)) {                                  
                muted_count += topic_count;                                                             
            } else if (sub.is_muted) {                                                                  
                muted_count += topic_count;                                                             
            } else {                                                                                    
                unmuted_count += topic_count;                                                           
            }                                                                                           
        }                                                                                               
        const stream_count = {                                                                          
            unmuted_count,                                                                              
            muted_count,                                                                                
            stream_is_muted: sub.is_muted,                                                              
        };                                                                                              
        return stream_count;                                                                            
    }                                                                                                   
```

I added this `TODO` comment to note the bug and we should aim to fix this in a follow-up; might be a good thing for you to pair with @prakhar1144 on to get to know each other a bit better, since he's the expert on how we handle unreads with muted streams.